app/assets/javascripts/dashboard/advanced_dataset/advanced_dataset_view.js

```diff
@@ -72,20 +73,35 @@ class AdvancedDatasetView extends React.PureComponent<Props, State> {
     );
 
     const { sortedInfo } = this.state;
+    const dataSourceSortedByRank = useLruRank
+      ? _.sortBy(filteredDataSource, ["lastUsedByUser", "created"]).reverse()
+      : filteredDataSource;
+
+    // Create a map from dataset to its rank
+    const datasetToRankMap: Map<DatasetType, number> = new Map(
+      dataSourceSortedByRank.map((dataset, rank) => [dataset, rank]),
+    );
+
     const sortedDataSource =
       // Sort using the dice coefficient if the table is not sorted otherwise
       // and if the query is longer then 3 characters to avoid sorting *all* datasets
       this.props.searchQuery.length > 3 && sortedInfo.columnKey == null
         ? _.chain(filteredDataSource)
-            .map(row => ({
-              row,
-              diceCoefficient: dice(row.name, this.props.searchQuery),
-            }))
-            .sortBy("diceCoefficient")
-            .map(({ row }) => row)
+            .map(dataset => {
+              const diceCoefficient = dice(dataset.name, this.props.searchQuery);
+              const rank = useLruRank ? datasetToRankMap.get(dataset) || 0 : 0;
+              const rankCoefficient = rank / filteredDataSource.length;
```

Comment:
As discussed, this should be `1 - (rank / filteredDataSource.length)` as the dice coefficient is 1 for two identical entities and, therefore, the rankCoefficient should also be 1 for the least recently used dataset.